packages/eslint-plugin-react-hooks/src/RulesOfHooks.js

```diff
@@ -522,6 +570,61 @@ export default {
           }
           reactHooks.push(node.callee);
         }
+
+        // Check if a function created with useEvent is invoked within
+        // a useEffect or closure.
+        const scope = context.getScope();
+        const variable = recursivelyFindVariableInScope(
+          scope,
+          node.callee.name,
+        );
+        if (variable) {
+          const isUseEvent = variable.defs.some(def => {
+            return isUseEventVariableDeclarator(def.node);
+          });
+          if (isUseEvent) {
+            let depth = 0;
+            for (const ancestor of context.getAncestors()) {
+              // For performance reasons we don't want to walk up
+              // every single ancestor node.
+              if (depth++ > MAX_USE_EVENT_ANCESTOR_DEPTH) {
+                break;
+              }
+              if (
+                ancestor.type === 'CallExpression' &&
+                ancestor.callee.name !== 'useEffect'
```

Comment:
I don't think this makes sense to me. From my understanding there's nothing special about `useEffect` here (but maybe I'm wrong?) E.g. I think you should be able to refactor `useEffect` to some random thing like `useIntersectionObserver` and that wouldn't break the rules.